src/main/java/com/chargerrobotics/utils/GyroHeading.java

```diff
@@ -0,0 +1,36 @@
+package com.chargerrobotics.utils;
+
+import java.nio.ByteBuffer;
+
+import com.chargerrobotics.utils.ArduinoSerialReceiver.ArduinoListener;
+
+public class GyroHeading extends ArduinoListener {
+
+    private static double heading = 0.0;
+    private static GyroHeading instance;
+    private static final Object lock = new Object();
```

Comment:
I already put a gyroscope receiver in the sensors package, but setting/getting the value of a variable is already thread-safe and doesn't need synchronization or anything. It should probably be volatile though to ensure the latest data is gotten by each thread.

Reply:
Actually I just realized it's a double and not an int, and if what I read was correct, setting a double isn't going to be thread-safe.